test_runner/src/main/kotlin/ftl/gc/android/CreateAndroidInstrumentationTest.kt

```diff
@@ -46,11 +48,19 @@ internal fun AndroidInstrumentationTest.setupTestTargets(
                     }
                     uniformSharding = UniformSharding().setNumShards(safeNumUniformShards)
                 } else {
-                    manualSharding = ManualSharding().setTestTargetsForShard(
-                        testShards.map {
-                            TestTargetsForShard().setTestTargets(it)
-                        }
-                    )
+                    manualSharding = if (testFlagForShard.isNotEmpty()) {
+                        ManualSharding().setTestTargetsForShard(
+                            testFlagForShard.map {
+                                TestTargetsForShard().setTestTargets(it.split(',', ';'))
+                            }
+                        )
+                    } else {
+                        ManualSharding().setTestTargetsForShard(
+                            testShards.map {
+                                TestTargetsForShard().setTestTargets(it)
+                            }
+                        )
+                    }
```

Comment:
How about ?

```kotlin
manualSharding = ManualSharding().setTestTargetsForShard(getTestShards())
...
fun getTestShards() = = if (testFlagForShard.isNotEmpty()) {
    testFlagForShard.map { TestTargetsForShard().setTestTargets(it.split(',', ';')) }
} else {
    testShards.map {TestTargetsForShard().setTestTargets(it) }
}
```

Reply:
nice cleanup. 👍

Reply:
unfortunately after several attempts it actually is very messy implementation due to all the requirements of the class. I will skip as it makes it more unreadable imho.